imap_processing/idex/idex_l1a.py

```diff
@@ -72,9 +73,52 @@ def __init__(self, packet_file: Union[str, Path]) -> None:
         -----
             Currently assumes one L0 file will generate exactly one L1a file.
         """
-        decom_packet_list = decom_packets(packet_file)
+        self.data = []
+        self.idex_attrs = get_idex_attrs()
+        epoch_attrs = self.idex_attrs.get_variable_attributes(
+            "epoch", check_schema=False
+        )
+
+        science_packets, datset_by_apid = decom_packets(packet_file)
+
+        if science_packets:
+            logger.info("Processing IDEX L1A Science data.")
+            self.data.append(self._create_science_dataset(science_packets))
+
+        elif IDEXAPID.IDEX_EVT in datset_by_apid:
```

Comment:
These should be bare "if"s because you'll be getting potentially all apids within a single file.
```suggestion
        if IDEXAPID.IDEX_EVT in datset_by_apid:
```